utils/src/main/java/software/amazon/awssdk/utils/ThreadFactoryBuilder.java

```diff
@@ -17,17 +17,25 @@
 
 import java.util.concurrent.Executors;
 import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 import software.amazon.awssdk.annotation.SdkProtectedApi;
+import software.amazon.awssdk.annotation.SdkTestInternalApi;
 
 /**
  * A builder for creating a thread factory. This allows changing the behavior of the created thread factory.
  */
 @SdkProtectedApi
 public class ThreadFactoryBuilder {
-    private static final AtomicInteger POOL_NUMBER = new AtomicInteger(1);
 
-    private String threadNamePrefix = "aws-java-sdk-thread";
+    /**
+     * To prevent thread names from becoming too long we limit to 4 digits for the pool number before wrapping back
```

Comment:
It seems unlikely there will be a clash given 10,000 thread names, not too mention many will/should probably be using more descriptive prefixes. And if there is a clash, it's for debugging purposes so it's not the end of the world.